src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -3017,7 +3017,8 @@ private void BindArrayInitializerExpressions(InitializerExpressionSyntax initial
         {
             bool hasErrors = false;
 
-            if (!IsInMethodBody && !IsLocalFunctionsScopeBinder)
+            if (!IsInMethodBody && !IsLocalFunctionsScopeBinder ||
+                !node.IsLegalSpanStackAllocPosition())
```

Comment:
Not related to this PR: should `IsLegalSpanStackAllocPosition` unwrap parenthesized expressions too?

Reply:
maybe, we will have to discuss that.

---
In reply to: [139545860](https://github.com/dotnet/roslyn/pull/22162#discussion_r139545860) [](ancestors = 139545860)